packages/pwa/app/commerce-api/contexts.js

```diff
@@ -0,0 +1,171 @@
+                } else {
+                    items.push(payload.item)
+                }
+                return {
+                    ...state,
+                    productLists: {
+                        ...state.productLists,
+                        [payload.listId]: {
+                            ...state.productLists[payload.listId],
+                            customerProductListItems: items
+                        }
+                    }
+                }
+            }
+            case CPLActionTypes.UPDATE_LIST_ITEM: {
+                const productLists = {
+                    ...state.productLists
+                }
+                productLists[payload.listId].customerProductListItems = productLists[
+                    payload.listId
+                ].customerProductListItems?.map((listItem) => {
+                    if (listItem.id === payload.item.id) {
+                        return {
+                            ...listItem,
+                            ...payload.item
+                        }
+                    }
+                    return listItem
+                })
+                return {...state, productLists}
+            }
+            case CPLActionTypes.REMOVE_LIST_ITEM: {
+                const productLists = {
+                    ...state.productLists
+                }
+                productLists[payload.listId].customerProductListItems = productLists[
+                    payload.listId
+                ].customerProductListItems?.filter((listItem) => {
+                    return listItem.id !== payload.itemId
+                })
+                return {...state, productLists}
+            }
+            case CPLActionTypes.SET_INITIALIZED: {
+                return {...state, isInitialized: payload}
+            }
+            case CPLActionTypes.RESET: {
+                return {...CPLInitialValue}
+            }
+            default:
+                throw new Error('Unknown action.')
+        }
+    }, CPLInitialValue)
+
+    const actions = {
+        receiveLists: (lists) => dispatch({type: CPLActionTypes.RECEIVE_LISTS, payload: lists}),
+        receiveList: (list) => dispatch({type: CPLActionTypes.RECEIVE_LIST, payload: list}),
+        createListItem: (listId, item) =>
+            dispatch({type: CPLActionTypes.CREATE_LIST_ITEM, payload: {listId, item}}),
+        updateListItem: (listId, item) =>
+            dispatch({type: CPLActionTypes.UPDATE_LIST_ITEM, payload: {listId, item}}),
+        removeListItem: (listId, itemId) =>
+            dispatch({type: CPLActionTypes.REMOVE_LIST_ITEM, payload: {listId, itemId}}),
+        setInitialized: (isInitialized) =>
+            dispatch({type: CPLActionTypes.SET_INITIALIZED, payload: isInitialized}),
+        reset: () => dispatch({type: CPLActionTypes.RESET})
+    }
+
+    return (
+        <_CustomerProductListsProvider value={{state, actions}}>
+            {children}
+        </_CustomerProductListsProvider>
```

Comment:
_CustomerProductListsProvider isn't being used anywhere else, is could be written as: 

```
<CustomerProductListsContext.Provider value={{state, actions}}>
    {children}
</CustomerProductListsContext.Provider>
```

I'm only saying this because using a `_` prefix is a strong style choice that I don't think we've been using much in the project. Maybe we try to make things look at homogenous as possible.

Reply:
👍 fixed